lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+        'ss_eye_pain',
+        'ss_fatigue',
+        'ss_fever',
+        'ss_gi',
+        'ss_headache',
+        'ss_hoarse',
+        'ss_myalgia',
+        'ss_nasal_drip',
+        'ss_nausea',
+        'ss_runny_nose',
+        'ss_sinus_pain',
+        'ss_skin_rash',
+        'ss_sleeping',
+        'ss_smell',
+        'ss_sneezing',
+        'ss_sob',
+        'ss_sore_throat',
+        'ss_sputum',
+        'ss_stomach_pain',
+        'ss_sweats',
+        'ss_taste',
+        'ss_vomiting',
+        'ss_wheeze',
+    ]
+
+    boolean_questions = [
+        'ss_nasal',
+        'ss_chest_symptoms',
+        'ss_sensory_symptoms',
+        'ss_eye_ear_throat',
+        'ss_gi',
+    ]
+
+    date_questions = [
+        'ss_date',
+    ]
+
+    coding_questions: List[str] = [
+    ]
+
+    question_categories = {
+        'valueInteger': integer_questions,
+        'valueBoolean': boolean_questions,
+        # 'valueString': string_questions, ## none so far
+        'valueDate': date_questions,
+        # 'valueCoding': coding_questions, ## none so far
+    }
+
+    return create_questionnaire_response(
+        record = record,
+        question_categories = question_categories,
+        patient_reference = patient_reference,
+        encounter_reference = encounter_reference,
+        system_identifier = INTERNAL_SYSTEM)
+
+
+def airs_create_computed_questionnaire_response(record: REDCapRecord, patient_reference: dict,
+    encounter_reference: dict, birthdate: datetime, encounter_date: datetime) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for a
+    computed questionnaire. This "computed questionnaire" produces
+    answers that do not appear directly in an actual REDCap survey.
+    For example, a computed question captures the participant's age
+    on the date of the encounter.
+    """
+    # A birthdate of None will return a falsy relativedelta() object
+    delta = relativedelta(encounter_date, birthdate)
+
+    if not delta:
+        age = None
+    else:
+        # Age Ceiling
+        age = age_ceiling(delta.years)
+
+    record['age'] = age
+
+    integer_questions = [
+        'age'
+    ]
+
+    question_categories = {
+        'valueInteger': integer_questions
+    }
+
+    return create_questionnaire_response(
+        record = record,
+        question_categories = question_categories,
+        patient_reference = patient_reference,
+        encounter_reference = encounter_reference,
+        system_identifier = INTERNAL_SYSTEM)
+
+
+def get_date_from_repeat_instance(instance_id: int) -> str:
+    """
+    Returns the date associated with a REDCap repeat instance.
+    This is safe to do only when the Daily Attestation exists
+    because the Musher sets the repeat instance based on the date.
+    Returns the date as a string because that is how it is used.
+    """
+    return (STUDY_START_DATE + relativedelta(days=(instance_id -1))).strftime('%Y-%m-%d')
```

Comment:
This was specific for HCT, and shouldn't be needed here.

Reply:
Noted and removed.